src/algebra/group/type_tags.lean

```diff
@@ -242,3 +242,14 @@ def add_monoid_hom.to_multiplicative'' [add_monoid α] [monoid β] :
 def monoid_hom.to_additive'' [add_monoid α] [monoid β] :
   (multiplicative α →* β) ≃ (α →+ additive β) :=
 add_monoid_hom.to_multiplicative''.symm
+
+/-- If `α` has some multiplicative structure and coerces to a function,
+  then `additive α` should also coerce to a function. -/
+instance additive.has_coe_to_fun {α : Type*} [has_coe_to_fun α] : has_coe_to_fun (additive α) := 
+⟨λ a, has_coe_to_fun.F a.to_mul, λ a, coe_fn a.to_mul⟩
+
+/-- If `α` has some additive structure and coerces to a function,
+  then `multiplicative α` should also coerce to a function. -/
+instance multiplicative.has_coe_to_fun {α : Type*} [has_coe_to_fun α] :
+  has_coe_to_fun (multiplicative α) := 
+⟨λ a, has_coe_to_fun.F a.to_add, λ a, coe_fn a.to_add⟩
```

Comment:
As suggested on Zulip, let's add `simps`
```suggestion
def has_coe_to_fun.simps.coe {α : Sort*} [has_coe_to_fun α] : Π a : α, has_coe_to_fun.F a :=
coe_fn

/-- If `α` has some multiplicative structure and coerces to a function,
  then `additive α` should also coerce to a function. -/
@[simps]
instance additive.has_coe_to_fun {α : Type*} [has_coe_to_fun α] : has_coe_to_fun (additive α) := 
⟨λ a, has_coe_to_fun.F a.to_mul, λ a, coe_fn a.to_mul⟩

@[simps]
/-- If `α` has some additive structure and coerces to a function,
  then `multiplicative α` should also coerce to a function. -/
instance multiplicative.has_coe_to_fun {α : Type*} [has_coe_to_fun α] :
  has_coe_to_fun (multiplicative α) := 
⟨λ a, has_coe_to_fun.F a.to_add, λ a, coe_fn a.to_add⟩
```
@fpvandoorn, where would you suggest we put `has_coe_to_fun.simps.coe`? Without it, the lemmas appear in an awkward form.

Reply:
Ok I'll add the simps, but the `has_coe_to_fin.simps.coe` line makes line 267 in data/equiv/mul_add.lean get stuck and timeout.

Reply:
Hmm, I don't think the simps is that useful without the `has_coe_to_fun.simps.coe` lemma too.

Reply:
I assume the line you're referring to is here?

https://github.com/leanprover-community/mathlib/blob/e6819d3a89e550615469e426dc4f2776068e9ef2/src/data/equiv/mul_add.lean#L267-L280

Reply:
Yeah that's the one